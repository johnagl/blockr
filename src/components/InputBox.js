import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import IconButton from '@material-ui/core/IconButton';
import AddIcon from '@material-ui/icons/Add';

const useStyles = makeStyles((theme) => ({
    root: {
      padding: '2px 4px',
      display: 'flex',
      alignItems: 'center',
      marginBottom: '10px'
    },
    input: {
      marginLeft: theme.spacing(1),
      flex: 1,
    },
    iconButton: {
      padding: 10,
    },
}));


const InputBox = ({handleAddWebsite, onSetUrlChange}) => {
    const classes = useStyles();
    return (
        <Paper component="form" className={classes.root}>
            <InputBase
                className={classes.input}
                placeholder="Website URL"
                onChange={onSetUrlChange}
            />
            <IconButton onClick={handleAddWebsite} className={classes.iconButton}>
                <AddIcon />
            </IconButton>
        </Paper>
    )
}

export default InputBox;