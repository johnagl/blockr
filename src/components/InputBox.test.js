import { shallow } from "enzyme";
import React from 'react';
import InputBox from './InputBox';

it('expect to render input box component', () => {
    expect(shallow(<InputBox />)).toMatchSnapshot();
})