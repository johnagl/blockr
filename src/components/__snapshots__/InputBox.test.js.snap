// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render input box component 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-root-1"
  component="form"
>
  <WithStyles(ForwardRef(InputBase))
    className="makeStyles-input-2"
    placeholder="Website URL"
  />
  <WithStyles(ForwardRef(IconButton))
    className="makeStyles-iconButton-3"
  >
    <Memo([object Object]) />
  </WithStyles(ForwardRef(IconButton))>
</WithStyles(ForwardRef(Paper))>
`;
